### PR TITLE
client: Add CreateMediaDownloadRequest API

### PR DESCRIPTION
**Pull request by superdump**

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -220,3 +220,29 @@ func (fc *Client) LookupServerKeys(
...
+
+	remoteReqAddr := "https://" + dnsResult.Addrs[0] + "/_matrix/media/v1/download/" + string(matrixServer) + "/" + mediaID
+	remoteReq, err := http.NewRequest("GET", remoteReqAddr, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	remoteReq.Header.Set("Host", string(matrixServer))
+
+	resp, err := fc.client.Do(remoteReq)
```
Comment (NegativeMjark) [inline client.go]: The http client in `fc.client` has been modified to support a "matrix://" URL scheme that handles the server lookup so all of the above code can be replaced with 

```go
requestURL := "matrix://" + string(matrixServer) + "/_matrix/media/v1/download/" + string(matrixServer) + "/" + mediaID
resp, err := fc.client.GET(requestURL)
```

File: client.go [hunk 2]
```diff
@@ -220,3 +220,29 @@ func (fc *Client) LookupServerKeys(
...
+	if err != nil {
+		return nil, fmt.Errorf("Failed to execute request for remote file")
```
Comment (NegativeMjark) [inline client.go]: I'd just return the `err` here.

File: client.go (hunk 1 above)
Comment (superdump) [inline client.go]: Done.

File: client.go (hunk 2 above)
Comment (superdump) [inline client.go]: Done.

**Pull request merged**